File: src/HZY.Services.Admin/DevelopmentTool/MonitorEFCore/Impl/MonitorEFCoreService.cs
using HzyEFCoreRepositories.Monitor;
using HzyEFCoreRepositories.Monitor.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HZY.Services.Admin.DevelopmentTool.MonitorEFCore.Impl
{
    /// <summary>
    /// 监控 EFCore 服务
    /// </summary>
    public class MonitorEFCoreService : IMonitorEFCoreService
    {
        public EFCoreMonitorContext GetEFCoreMonitorContext()
        {
            return MonitorEFCoreCache.Context;
        }

        public List<EFCoreMonitorSqlContext> GetNewest()
        {
            return MonitorEFCoreCache.SqlContext.OrderByDescending(w => w.Time).Take(20).ToList();
        }

        public List<EFCoreMonitorSqlContext> GetTimeConsumingRanking()
        {
            return MonitorEFCoreCache.SqlContext.OrderByDescending(w => w.ElapsedMilliseconds).Take(20).ToList();
        }



    }
}

File: hzy-admin-client-element-plus/src/services/development_tool/MonitorEfCoreService.ts
import Http from '@/infrastructure/scripts/Http';

const controllerName = "admin/MonitorEFCore";

export default {
    /**
     * 获取 EFCore 监控上下文
     *
     */
    getEFCoreMonitorContext() {
        return Http.post(`${controllerName}/GetEFCoreMonitorContext/`, null, false);
    },
    /**
     * 获取 EFCore Sql 监控上下文 耗时排行榜
     *
     */
    getTimeConsumingRanking() {
        return Http.post(`${controllerName}/GetTimeConsumingRanking/`, null, false);
    },
    /**
    * 获取 EFCore Sql 监控上下文 最新数据
    *
    */
    getNewest() {
        return Http.post(`${controllerName}/GetNewest/`, null, false);
    },

};